js/src/utils/paid-ads/validateForm.js

```diff
@@ -25,7 +25,7 @@ const validateForm = ( values ) => {
 		);
 	}
 
-	if ( values.amount <= 0 ) {
+	if ( ! Number.isFinite( values.amount ) || values.amount <= 0 ) {
```

Comment:
❓  How comes `amount` could be a non finite number?

Reply:
The budget will temporarily be set to `undefined` when the input is disabled.

https://github.com/woocommerce/google-listings-and-ads/blob/c08364ff753c5a2c50096e5b2254fd0c9ded0032/js/src/components/paid-ads/budget-section/index.js#L41-L50